pkgs/misc/vim-plugins/update.py

```diff
@@ -477,13 +482,27 @@ def parse_args():
         default=DEFAULT_OUT,
         help="Filename to save generated nix code",
     )
+    parser.add_argument(
+        "--commit",
+        dest="commit",
+        action="store_true",
+        help="Automatically commit updates",
+    )
 
     return parser.parse_args()
 
 
-def main() -> None:
+def get_nixpkgs_repo() -> git.Repo:
+    repo = git.Repo(NIXPKGS_PATH)
+    if repo.is_dirty():
+        raise Exception("Please stash your changes before updating.")
```

Comment:
This may be confusing to a git novice. How about "You have uncommitted changes. Please commit your changes or stash them before running this tool."